src/services/AccountService.ts

```diff
@@ -107,6 +116,37 @@ export async function getAccountBalancesInfo(accountId: string): Promise<Account
     }
 }
 
+export async function getAccountBalancesSummary(accountId: string): Promise<AccountBalancesSummary> {
+    let accountBalancesInfo = await getAccountBalancesInfo(accountId);
+    console.log(accountBalancesInfo);
+
+    let totalAvgPaidPrice = new Big("0");
+    let totalOutcomePrice = 0;
+    let spent = 0;
+    let outcomeTokens = 0;
+
+    for (let i = 0; i < accountBalancesInfo.marketBalances.length; i++) {
+        totalAvgPaidPrice = accountBalancesInfo.marketBalances[i].avgPaidPrice.add(totalAvgPaidPrice);
+        totalOutcomePrice += accountBalancesInfo.marketBalances[i].outcomePrice;
+        spent += Number(accountBalancesInfo.marketBalances[i].spent);
+        outcomeTokens += Number(accountBalancesInfo.marketBalances[i].balance);
+    }
+
+    const unrealizedPnl = totalAvgPaidPrice.gt("0") ? new Big(totalOutcomePrice).minus(totalAvgPaidPrice).div(totalAvgPaidPrice).mul(100).round(2) : new Big("0");
+    const totalSpent = String(spent);
+    const outcomeTokenBalance = String(outcomeTokens);
+    // assume all collateral tokens are the same as the first for master balance
+    // TODO: account for multiple collateral tokens in total balance??
```

Comment:
Yeah, that would be better. If you use the TokenViewModel you can get the price of the collateral. This would allow you to calculate all the balances in dollars.